operations/iuf/stages/management_nodes_rollout.md

```diff
@@ -26,22 +35,184 @@ The following arguments are most often used with the `management-nodes-rollout`
+documentation to get the values for `final_image_id` and `configuration` for images with a `configuration_group_name` value matching `Management_Worker`. These values will be used in the next step.
+
+1. Upgrade/rebuild the worker node. The worker node is automatically rebuilt using Argo workflows. If rebuilding multiple worker nodes at once, see [this page](../../node_management/Rebuild_NCNs/Rebuild_NCNs.md#restrictions) for restrictions.
+
+    (`ncn-m001#`) Rebuild a worker node. Use the values acquired in the previous step in place of `<final_image_id>` and `<configuration>`.
+
+    ```bash
+    /usr/share/doc/csm/upgrade/scripts/upgrade/ncn-upgrade-worker-storage-nodes.sh ncn-w001 --image-id <final_image_id> --desired-cfs-conf <configuration>
+    ```
+
+## Action needed if a worker rebuild fails
+
+In general, worker node rebuilds should complete successfully before starting another rebuild.
+A worker node can get into a bad state if it has been partially rebuilt and an attempt is made to restart the rebuild on that same node.
+As a result, it is not possible to start another worker node rebuild if there is an existing incomplete worker node rebuild workflow, where "incomplete" means it has stopped before successfully completing the full workflow.
+If an incomplete workflow exists and an attempt is made to start another worker rebuild workflow,
+the original incomplete worker rebuild workflow will continue and no new workflow will be created.
 
-When NCN management nodes are being rebuilt, an additional Argo workflow will execute in addition to the standard IUF Argo workflows. It will include the string `ncn-lifecycle-rebuild` in its name.
+If it is necessary to start an entirely new worker rebuild workflow after a previous worker rebuild workflow failed, the failed workflow must be deleted from Kubernetes first.
 
-## Example
+**`WARNING`** Deleting a workflow will delete information about the state of that workflow and the steps that have been completed.
+Deleting a partially complete workflow should be done cautiously and only if needed.
+
+To delete a failed Argo workflow, complete the following steps.
+
+1. Get the name of the failed workflow. All worker rebuild workflows start with `ncn-lifecycle-rebuild`. The name of the worker rebuild workflow can be found in the Argo UI or by searching workflows in Kubernetes with the following command.
+
+    (`ncn-m#`) List failed worker rebuild workflows. Note that NCN storage upgrade workflows will also contain `ncn-lifecycle-rebuild` in their name and may be present in this list.
+
+    ```bash
+    kubectl get workflows -n argo | grep 'ncn-lifecycle-rebuild' | grep 'Fail'
+    ```
+
+1. (`ncn-m#`) Delete the failed workflow.
+
+    ```bash
+    kubectl delete workflows -n argo <failed workflow>
```

Comment:
Is this different than deleting a workflow from the Argo UI?

Reply:
@atifsyedali do you know the answer to this?  @leliasen-hpe wrote this part and I am not sure.

Reply:
Deleting an workflow from the Argo UI is disabled. So one must use the command line to explicitly delete this workflow.

Reply:
I will leave this as is per Atif's comment.  If there still concern here, let me know.